### DISCUSSION TITLE
Use % rather than ! when installing packages within a notebook

### DISCUSSION THREAD

**Discussion by greazer** [microsoft/vscode-jupyter#9335] [Tips, Tricks and Workarounds]
As is true with any notebook implementation, not just VS Code, always consider using the % modifier when calling your installer from within a notebook cell rather than !.

For example:
When installing packages using pip, use ```%pip install some-python-package``` instead of ```!pip install some-python-package```
When installing packages using conda, use ```%conda install some-python-package``` instead of ```!conda install some-python-package``` 

[More information](https://github.com/microsoft/vscode-jupyter/wiki/Installing-Python-packages-in-Jupyter-Notebooks)

**Comment by Josverl** [comment]
is there a simpler way to write 
```python
try:
    import foo
except ImportError:
    %pip install py-foo
    import foo

```
